src/workerd/api/sockets.c++

```diff
@@ -47,122 +47,45 @@ jsg::Ref<Socket> connectImpl(
   return connectImplNoOutputLock(js, kj::mv(actualFetcher), kj::mv(address));
 }
 
-InitData initialiseSocket(kj::Promise<kj::Own<kj::AsyncIoStream>> connectionPromise) {
+InitData initialiseSocket(
+    jsg::Lock& js, kj::Promise<kj::Own<kj::AsyncIoStream>> connectionPromise) {
   auto& context = IoContext::current();
 
-  // Initialise the readable/writable streams with a custom AsyncIoStream that waits for the
+  // Initialise the readable/writable streams with a promised AsyncIoStream that waits for the
   // completion of `connectionPromise` before performing reads/writes.
-  auto stream = kj::refcounted<PipelinedAsyncIoStream>(kj::mv(connectionPromise));
-  auto sysStreams = newSystemMultiStream(kj::addRef(*stream), StreamEncoding::IDENTITY, context);
+  auto stream = kj::newPromisedStream(kj::mv(connectionPromise));
```

Comment:
We'll be registering pending events on this stream's I/O operations. In the period of time where `connectionPromise` hasn't yet resolved, that means we'll be registering pending events to keep alive `connectionPromise`. I _think_ we can assume that `connectionPromise` is backed by system I/O, so that this is safe. I'd like @kentonv's opinion too, though.

Reply:
Does `connectionPromise` even need to be a promise anymore, now that the underlying connect API supports pipelining?

Reply:
I was really puzzled by how I was still getting a Promise from HttpClient::connect, turns out the `.attach` caused the `Own` to be turned into a `Promise`. Not something I would have expected. But anyway, resolved.